### PR TITLE
ZelNodes

### PR DESCRIPTION
**Pull request by TheTrunk**

### CODE REVIEW

File: contrib/debian/changelog [hunk 1]
```diff
@@ -1,3 +1,9 @@
+zelcash (3.0.0) stable; urgency=medium
+
+  * 3.0.0 release.
+
+ -- Zelcash Company <team@zel.cash>  Mon, 18 Jan 2019 12:00:00 +0100
```
Comment (Scribbles-MCAMK) [inline contrib/debian/changelog]: Incorrect contact info and date?

File: src/config/bitcoin-config.h [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+#define LT_OBJDIR ".libs/"
+
+/* Define to the address where bug reports for this package should be sent. */
+#define PACKAGE_BUGREPORT "https://github.com/zelcash/zelcash/issues"
+
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "Zelcash"
+
+/* Define to the full name and version of this package. */
+#define PACKAGE_STRING "Zelcash 1.1.0"
```
Comment (Scribbles-MCAMK) [inline src/config/bitcoin-config.h]: Is this correct version?

File: src/config/bitcoin-config.h [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+
+/* Define to the one symbol short name of this package. */
+#define PACKAGE_TARNAME "zelcash"
+
+/* Define to the home page for this package. */
+#define PACKAGE_URL ""
+
+/* Define to the version of this package. */
+#define PACKAGE_VERSION "1.1.0"
```
Comment (Scribbles-MCAMK) [inline src/config/bitcoin-config.h]: Correct Version?

File: README.md [hunk 4]
```diff
@@ -42,20 +41,16 @@ cd zelcash
 ```
 
 #### Run ZelCash 
-1. Create zelcash.conf file
+1. Create zelcash.conf file (copy and paste this block in one into your terminial)
```
Comment (Lumiboy) [inline README.md]:
```suggestion
1. Create zelcash.conf file (copy and paste this block in one into your terminal)
```

File: src/zelnode/zelnodeconfig.cpp [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    int linenumber = 1;
+    boost::filesystem::path pathZelnodeConfigFile = GetZelnodeConfigFile();
+    boost::filesystem::ifstream streamConfig(pathZelnodeConfigFile);
+
+    if (!streamConfig.good()) {
+        FILE* configFile = fopen(pathZelnodeConfigFile.string().c_str(), "a");
+        if (configFile != NULL) {
+            std::string strHeader = "# Zelnode config file\n"
+                                    "# Format: alias IP:port zelnodeprivkey collateral_output_txid collateral_output_index\n"
+                                    "# Example: zn1 127.0.0.2:16125 93HaYBVUCYjEMeeH1Y4sBGLALQZE1Yc1K64xiqgX37tGBDQL8Xg 2bcd3c84c84f87eaa86e4e56834c92927a07f9e18718810b92e0d0324456a67c 0\n";
```
Comment (Scribbles-MCAMK) [inline src/zelnode/zelnodeconfig.cpp]: Maybe use an ip that looks nothing like localhost?

File: src/config/bitcoin-config.h (hunk 3 above)
Comment (Scribbles-MCAMK) [inline src/config/bitcoin-config.h]: Correct version?

File: src/config/bitcoin-config.h (hunk 2 above)
Comment (Scribbles-MCAMK) [inline src/config/bitcoin-config.h]: Correct Version?

File: contrib/debian/changelog (hunk 1 above)
Comment (Scribbles-MCAMK) [inline contrib/debian/changelog]: Date and info is incorrect.

File: src/zelnode/zelnodesync.cpp [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+    if (fImporting || fReindex) return false;
+
+    TRY_LOCK(cs_main, lockMain);
+    if (!lockMain) return false;
+
+    CBlockIndex* pindex = chainActive.Tip();
+    if (pindex == NULL) return false;
+
+
+    // TODO, remove this constraint after testing on TESTNET which doesn't have a solid block schedule
```
Comment (Scribbles-MCAMK) [inline src/zelnode/zelnodesync.cpp]: Still needs to be done? Don't see any harm with it being left in here, but for the sake of cleanliness.

Comment (TheTrunk) [reply src/zelnode/zelnodesync.cpp]: This is basically saying: If the last block we have in database is 2+ days old, we are not synced. We may lower this time but this is not harmful for now.

**Pull request merged**